### PR TITLE
Week 3 chris frey

### PR DESCRIPTION
**Pull request by Chris-Frey**
# Submitting Week 3 Assessment

### ✅ Completed Files

Select the following as confirmation that the files in this repository are complete.

- [ ] Coding Challenges
- [ ] Coding Concepts
- [ ] Interview Questions

### ✋ Reminder

Do NOT merge your assessment!

### 📝 Feedback
I'm not sure which order the intent is, but the coding challenge file makes it seem like we're supposed to solve the coding problem and then, after, do the pseudo-code.  My understanding is it should be the other way around.

If you would like specific feedback please leave details here:

### CODE REVIEW

File: code-challenges.test.js [hunk 1]
```diff
@@ -21,13 +27,32 @@ const fibLength2 = 10
...
+  return arr
+}
+fibIt(fibLength1)
+fibIt(fibLength2)
 // Pseudo code:
+// create arrow function with one parameter
+// define variable with array containing [1, 1] because those are always in the fib array
+// create for loop 
+// add the value of the current index positions -1, and -2 and push them to the existing array
+// return array
```
Comment (chrisyhlee) [inline code-challenges.test.js:43]: Great logic and great pseudo code!

File: code-challenges.test.js [hunk 2]
```diff
@@ -69,5 +108,23 @@ const accountTransactions3 = []
...
+    arr.push(input[i] + arr[i - 1])
+  }
+  
+  return arr
+}
 
-// Pseudo code:
+  sumIt(accountTransactions1)
+  sumIt(accountTransactions2)
+  sumIt(accountTransactions3)
```
Comment (chrisyhlee) [inline code-challenges.test.js]: I would remove these function calls for all of the questions, as you are already performing them in your tests.

File: code-challenges.test.js [hunk 3]
```diff
@@ -69,5 +108,23 @@ const accountTransactions3 = []
 // Expected output: []
 
 // b) Create the function that makes the test pass.
+const sumIt = (input) => {
+  let arr = []
+  arr.push(input[0])
+  for(let i = 1; i < input.length ; i++) {
+    console.log(arr);
```
Comment (chrisyhlee) [inline code-challenges.test.js]: Be sure to remove console.logs when you're done using them. Nice work separating the accumulated sum from the input array

File: interview-questions.md [hunk 4]
```diff
@@ -8,48 +8,50 @@ Then, research the question to expand on your answer. Even if you feel you have
...
+Researched answer: JSX converts HTML tags into React elements. Though I already knew this, I suppose it would have been helpful to add: using `{}` allows us to write javascript code inside HTML elements.  All code that goes into JSX is intended to be transformed into JavaScript.
 
-1. What is yarn? What file(s) are modified when you run the command yarn in your terminal?
+2. What is yarn? What file(s) are modified when you run the command yarn in your terminal?
 
-Your answer:
+Your answer: Yarn is a package management tool for React. it can be used to install and run things like the "Jest" debugging software.
 
-Researched answer:
+Researched answer: Yarn is allows you to share code with others.
```
Comment (chrisyhlee) [inline interview-questions.md:19]: Yarn will modify the yarn.lock file, and may also modify the package.json file.

File: interview-questions.md [hunk 5]
```diff
@@ -8,48 +8,50 @@ Then, research the question to expand on your answer. Even if you feel you have
...
 
-Researched answer:
+Researched answer: Components are reusable pieces of code, akin to functions in JavaScript. They can be passed as props to other components.
 
 4. What is the difference between state values and props in React?
 
-Your answer:
+Your answer: props are essentially more advanced parameters that are used in component files.  They're used as vehicles to pass specific values from `App.js` to the components, and get a return.  These values are only precessed when needed, and are not held in a persistent capacity, and the value passing only happens in the downstream direction (away from `App.js`)
```
Comment (chrisyhlee) [inline interview-questions.md:29]: It's important to know that it is not only from App.js to other components. Props can be passed from any component, as long as there is a nested component. Like you said, the value passing happens downstream for conventional props, but props being passed from any component to a nested component within that component is still downstream!

File: interview-questions.md [hunk 6]
```diff
@@ -8,48 +8,50 @@ Then, research the question to expand on your answer. Even if you feel you have
...
+  State values are assigned to a state variable, which holds any values it has until they are overwritten.  The information in a state variable is what is passed through the 'prop' in a component file, to run the code in that component.
+  
 
-Researched answer:
+Researched answer: State is mutable, while props are immutable. Meaning the value being passed to props can change, but a prop cannot change the property once it has been passed. It seems that from a performance perspective, it is better to work with props, than just state.
```
Comment (chrisyhlee) [inline interview-questions.md:33]: Good addition to your answer with the mutable and immutable

**Review by chrisyhlee — APPROVED** [review]
Nice work, Chris! You are officially halfway done with assessments! This currently displays Competent, as you definitely have a great understanding of the concepts you are using and are able to not only select what tools you need, but also use them in your own way for your own solutions. I left some comments throughout for you to read through. The main focus for you would be to clean up your files, as in unnecessary function calls and console.logs left in. Also, you're absolutely right about the positioning of the pseudo code sections, pseudo code should be written first. I'll look into that!

File: interview-questions.md (hunk 5 above)
Comment (Chris-Frey) [reply interview-questions.md:29]: Thanks for that clarification. Updated my mental model with that info!

File: interview-questions.md (hunk 4 above)
Comment (Chris-Frey) [reply interview-questions.md:19]: Understood. I think I need to read up more on yarn in general.  Thank you 🙂

**Pull request merged**